Add a note in spec about prefetch method execution (#1572)
ballerina-platform/module-ballerina-graphql: docs/spec/spec.md

The user is responsible for implementing the logic to collect the keys of the data to be loaded into the `DataLoader` in the `preFoo` method. Subsequently, the user can implement the logic to retrieve the data from the `DataLoader` within the `foo` method.

> >**Note:** If there is a remote method and a resource method with the same name, such as `bar`, and there exists a prefetch method named `preBar`, the GraphQL Engine will execute the `preBar` method prior to executing each `bar` method. Users can modify this behavior by utilizing the `prefetchMethodName` configuration within the `@graphql:ResourceConfig` annotation.

Reviewer: ```suggestion
>**Note:** If there multiple `resource`/`remote` methods with the same name in a GraphQL service, such as `bar`, and there exists a prefetch method named `preBar`, the GraphQL Engine will execute the `preBar` method prior to executing each `bar` method. Users can modify this behavior by utilizing the `prefetchMethodName` configuration within the `@graphql:ResourceConfig` annotation.
```
